### PR TITLE
state rejection reasons

### LINKED ISSUE #119: Add rejection reasons for state change
### Description

Oftentimes state changes are rejected.

### Acceptance Criteria

DO it.  Add JSON for it.

### Proposed Solution

Add a CAN message for why they were rejected, using a rejection code enum or something.

### LINKED ISSUE #330: Add rejection reasons for state change
### Description

Oftentimes state changes are rejected.

### Acceptance Criteria

DO it.  Add JSON for it.

### Proposed Solution

Add a CAN message for why they were rejected, using a rejection code enum or something.

### PR DESCRIPTION
**Pull request by chpy04**
## Changes

Added an enum for the different reasons a state change can be rejected. Sends 8 bits over can representing the 8 reasons (bc more than 1 can be true at the same time)

## Notes

Wasn't really sure what to put for the sim so I just added the interpretation as if it were a number although obv that doesn't make sense.

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes Northeastern-Electric-Racing/Cerberus-2.0#119

### CODE REVIEW

File: Core/Inc/state_machine.h [hunk 1]
```diff
@@ -38,6 +38,17 @@ typedef enum {
 	MAX_NERO_STATES
 } nero_menu_t;
 
+typedef enum {
+	NO_ERROR,
```
Comment (caiodasilva2005) [inline Core/Inc/state_machine.h]: nitpick: can you change to like `START_TRANSITION_OK` or thing like that

File: Core/Inc/state_machine.h [hunk 2]
```diff
@@ -38,6 +38,17 @@ typedef enum {
...
+	REVERSE_DISABLED,
+	DRIVE_FROM_FAULT,
+	ENTER_DRIVE_DISABLED,
+	ENTER_DRIVE_TSMS_OFF,
+	ENTER_DRIVE_BREAKS_NOT_ENGAGED,
+	ENTER_GAMES_TSMS_ON,
+	ENTER_GAMES_WHILE_MOVING
+} state_transition_error_t;
```
Comment (caiodasilva2005) [inline Core/Inc/state_machine.h:50]: just so you dont have to it bitshift these in the code later you coulddo it like this:
```
typedef enum {
    ERROR_OK                       = 0,
    REVERSE_DISABLED               = 1 << 0,
    DRIVE_FROM_FAULT               = 1 << 1,
    ENTER_DRIVE_DISABLED           = 1 << 2,
    ENTER_DRIVE_TSMS_OFF           = 1 << 3,
    ENTER_DRIVE_BREAKS_NOT_ENGAGED = 1 << 4,
    ENTER_GAMES_TSMS_ON            = 1 << 5,
    ENTER_GAMES_WHILE_MOVING       = 1 << 6
} state_transition_error_t;
```
Then you can just OR them together with no bit shift

File: Core/Src/state_machine.c [hunk 3]
```diff
@@ -119,6 +120,7 @@ static int transition_functional_state(func_state_t new_state, pdu_t *pdu,
 	case F_REVERSE:
 #ifdef DISABLE_REVERSE
 		printf("Reverse is disabled.");
+		cerberus_state.transition_error = (1 << REVERSE_DISABLED);
```
Comment (caiodasilva2005) [inline Core/Src/state_machine.c:123]: with the above comment, then just OR `REVERSE_DISABLED` into the transition_error byte with no bit shift

File: Core/Src/state_machine.c [hunk 4]
```diff
@@ -132,22 +134,36 @@ static int transition_functional_state(func_state_t new_state, pdu_t *pdu,
 		    (!brake_state ||
 		     cerberus_state.functional ==
 			     FAULTED)) { // only enforce brake / fault if tsms is actually on
+			if (!brake_state) {
+				cerberus_state.transition_error |= (1 << ENTER_DRIVE_BREAKS_NOT_ENGAGED);
+			}
+			if (cerberus_state.functional == FAULTED) {
+				cerberus_state.transition_error |= (1 << DRIVE_FROM_FAULT);
```
Comment (caiodasilva2005) [inline Core/Src/state_machine.c]: same here

**Review by caiodasilva2005 — CHANGES_REQUESTED** [review]
